pennylane/ops/qubit/parametric_ops.py

```diff
@@ -837,18 +857,18 @@ def simplify(self):
         Hadamard(wires=[0])
 
         """
-        p0, p1, p2 = np.mod(self.data, 2 * np.pi)
+        p0, p1, p2 = [p % (2 * np.pi) for p in self.data]
 
-        if np.allclose(p0, np.pi / 2) and np.allclose(np.mod(self.data[2], -2 * np.pi), -np.pi / 2):
-            return qml.RX(self.data[1], wires=self.wires)
+        if np.allclose(p0, np.pi / 2) and np.allclose(p2, 3 * np.pi / 2):
```

Comment:
```suggestion
        if qml.math.allclose(p0, np.pi / 2) and qml.math.allclose(p2, 3 * np.pi / 2):
```
Much nicer comparing it to `3*np.pi/2` +1

Reply:
Already changed in previous commit 👍